app/src/main/java/com/infomaniak/mail/ui/main/newMessage/NewMessageViewModel.kt

```diff
@@ -150,21 +154,58 @@ class NewMessageViewModel(application: Application) : AndroidViewModel(applicati
             draft.bcc = mailBcc.toRealmList()
             draft.subject = mailSubject
             draft.body = mailBody
+            draft.attachments = mailAttachments.toRealmList()
             draft.action = action
         }
     }
 
+    fun importAttachments(uris: List<Uri>) = viewModelScope.launch(Dispatchers.IO) {
+        val newAttachments = mutableListOf<Attachment>()
+        var attachmentsSize = mailAttachments.sumOf { it.size }
+
+        uris.forEach { uri ->
+            val availableSpace = FILE_SIZE_25_MB - attachmentsSize
+            val (attachment, hasSizeLimitBeenReached) = importAttachment(uri, availableSpace)
+
+            if (hasSizeLimitBeenReached) {
+                importedAttachments.postValue(newAttachments to ImportationResult.FILE_SIZE_TOO_BIG)
+                return@launch
+            }
+
+            attachment?.let {
+                newAttachments.add(it)
+                attachmentsSize += it.size
+            }
+        }
+
+        importedAttachments.postValue(newAttachments to ImportationResult.SUCCESS)
+    }
+
+    private fun importAttachment(uri: Uri, availableSpace: Int): Pair<Attachment?, Boolean> {
+        val (fileName, fileSize) = uri.getFileNameAndSize(getApplication())!!
+        if (fileSize > availableSpace) return null to true
+
+        return LocalStorageUtils.copyDataToAttachmentsCache(getApplication(), uri, fileName, currentDraftLocalUuid)?.let { file ->
+            val fileExtension = file.path.substringAfterLast(".")
+            val mimeType = MimeTypeMap.getSingleton().getMimeTypeFromExtension(fileExtension) ?: "*/*"
```

Comment:
A method exists in kDrive, so put in Core at this time
https://github.com/Infomaniak/android-kDrive/blob/d56ee5599005ce97521d79207d3522e2839d1638/app/src/main/java/com/infomaniak/drive/data/models/File.kt#L307